modules/abstract-cosmos/src/lib/transaction.ts

```diff
@@ -174,64 +172,65 @@ export class CosmosTransaction extends BaseTransaction {
     switch (json.type) {
       case TransactionType.Send:
         explanationResult.type = TransactionType.Send;
-        message = json.sendMessages[0].value as SendMessage;
-        outputAmount = message.amount[0].amount;
-        outputs = [
-          {
-            address: message.toAddress,
-            amount: outputAmount,
-          },
-        ];
+        outputAmount = BigInt(0);
+        outputs = json.sendMessages.map((message) => {
+          const sendMessage = message.value as SendMessage;
+          outputAmount = (outputAmount as bigint) + BigInt(sendMessage.amount[0].amount);
+          return {
+            address: sendMessage.toAddress,
+            amount: sendMessage.amount[0].amount,
+          };
+        });
         break;
       case TransactionType.StakingActivate:
         explanationResult.type = TransactionType.StakingActivate;
-        message = json.sendMessages[0].value as DelegateOrUndelegeteMessage;
-        outputAmount = message.amount.amount;
-        outputs = [
-          {
-            address: message.validatorAddress,
-            amount: outputAmount,
-          },
-        ];
+        outputAmount = BigInt(0);
+        outputs = json.sendMessages.map((message) => {
+          const delegateMessage = message.value as DelegateOrUndelegeteMessage;
+          outputAmount = (outputAmount as bigint) + BigInt(delegateMessage.amount.amount);
+          return {
+            address: delegateMessage.validatorAddress,
+            amount: delegateMessage.amount.amount,
+          };
+        });
         break;
       case TransactionType.StakingDeactivate:
         explanationResult.type = TransactionType.StakingDeactivate;
-        message = json.sendMessages[0].value as DelegateOrUndelegeteMessage;
-        outputAmount = message.amount.amount;
-        outputs = [
-          {
-            address: message.validatorAddress,
-            amount: outputAmount,
-          },
-        ];
+        outputAmount = BigInt(0);
+        outputs = json.sendMessages.map((message) => {
+          const delegateMessage = message.value as DelegateOrUndelegeteMessage;
+          outputAmount = (outputAmount as bigint) + BigInt(delegateMessage.amount.amount);
+          return {
+            address: delegateMessage.validatorAddress,
+            amount: delegateMessage.amount.amount,
+          };
+        });
         break;
       case TransactionType.StakingWithdraw:
         explanationResult.type = TransactionType.StakingWithdraw;
-        message = json.sendMessages[0].value as WithdrawDelegatorRewardsMessage;
-        outputs = [
-          {
-            address: message.validatorAddress,
+        outputs = json.sendMessages.map((message) => {
+          const withdrawMessage = message.value as WithdrawDelegatorRewardsMessage;
+          return {
+            address: withdrawMessage.validatorAddress,
             amount: UNAVAILABLE_TEXT,
-          },
-        ];
-        outputAmount = UNAVAILABLE_TEXT;
```

Comment:
was this intentionally removed ?

Reply:
yes, to make is similar to atom. I also checked that it is not being used anywhere